jenkins-master/jobs/scripts/workspace/runtests.py

```diff
@@ -4,231 +4,123 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import optparse
+import argparse
 import os
+import shutil
 import subprocess
 import sys
-import uuid
 
 here = os.path.dirname(os.path.abspath(__file__))
+venv_path = os.path.join(here, 'tests_venv')
 
 
-class Runner(object):
-
-    def __init__(self, venv_path):
-        self.venv_path = venv_path
-        self.build_path = os.path.join(here, 'build')
-
-        self.s3_bucket = None
-
-        os.chdir(here)
-        self.activate_venv()
+# Create the test environment and activate it
+# TODO remove once we make use of the mozharness script
+import environment
+command = ['python', os.path.join(here, 'firefox-ui-tests', 'create_venv.py'),
+           '--strict', '--with-optional', venv_path]
+print('Run command to create virtual environment for Firefox UI Tests: %s' % command)
+subprocess.check_call(command)
+environment.activate(venv_path)
 
-    def activate_venv(self):
-        os.chdir(os.path.join(here, 'firefox-ui-tests'))
-        command = ['python', 'create_venv.py', self.venv_path]
+# Can only be imported after the environment has been activated
+import firefox_ui_tests
+import firefox_puppeteer
 
-        print('Calling command to create virtual environment: %s' % command)
-        subprocess.check_call(command)
+from mozdownload import FactoryScraper
```

Comment:
This is in runtests which creates the environment via https://github.com/mozilla/firefox-ui-tests/blob/mozilla-central/create_venv.py. All this will be removed with mozharness.